src/Bicep.LangServer/Completions/ModuleReferenceCompletionProvider.cs

```diff
@@ -0,0 +1,574 @@
+                                return $"{modulePath}/{filePath}";
+                            }
+                        }
+                        else
+                        {
+                            if (filePath.StartsWith("bicep/"))
+                            {
+                                return filePath.Substring("bicep/".Length);
+                            }
+                        }
+                    }
+                }
+            }
+
+            return null;
+        }
+
+        private ImmutableSortedDictionary<string, OciArtifactModuleAlias> GetOciArtifactModuleAliases(Uri templateUri)
+        {
+            var rootConfiguration = configurationManager.GetConfiguration(templateUri);
+            return rootConfiguration.ModuleAliases.GetOciArtifactModuleAliases();
+        }
+
+        // Handles path completions.
+        private async Task<IEnumerable<CompletionItem>> GetPathCompletions(BicepCompletionContext context, string replacementText, Uri templateUri)
+        {
+            if (!IsOciModuleRegistryReference(replacementText))
+            {
+                return Enumerable.Empty<CompletionItem>();
+            }
+
+            if (replacementText == "'br/public:'" ||
+                replacementText == $"'br:{MCRRegistry}/bicep/'" ||
+                replacementText == "'br/public:" ||
+                replacementText == $"'br:{MCRRegistry}/bicep/")
+            {
+                return await GetPublicMCRPathCompletions(replacementText, context);
+            }
+            else
+            {
+                List<CompletionItem> completions = new List<CompletionItem>();
+
+                completions.AddRange(GetACRPartialPathCompletionsFromBicepConfig(replacementText, context, templateUri));
+                completions.AddRange(await GetMCRPathCompletionFromBicepConfig(replacementText, context, templateUri));
+
+                return completions;
+            }
+        }
+
+
+        // Handles path completions for case where user has specified an alias in bicepconfig.json with registry set to "mcr.microsoft.com".
+        private async Task<IEnumerable<CompletionItem>> GetMCRPathCompletionFromBicepConfig(string replacementText, BicepCompletionContext context, Uri templateUri)
+        {
+            List<CompletionItem> completions = new List<CompletionItem>();
+
+            var replacementTextWithTrimmedEnd = replacementText.TrimEnd('\'');
+
+            if (IsAcrRegistryReference(replacementTextWithTrimmedEnd, out _))
+            {
+                return completions;
+            }
+
+            foreach (var kvp in GetOciArtifactModuleAliases(templateUri))
+            {
+                if (kvp.Value.Registry is string registry)
+                {
+                    // We currently don't support path completion for ACR, but we'll go ahead and log telemetry to track usage.
+                    if (registry.EndsWith("azurecr.io", StringComparison.Ordinal) &&
```

Comment:
>"azurecr.io [](http://example.com/codeflow?start=42&length=11)

Another location with azurecr.io.

Reply:
Thanks! Removed it here:
https://github.com/Azure/bicep/commit/5a344b5a7897d42fc321ea1f024a6fdf4b392c1f